### DISCUSSION TITLE
Improve the CLI _pager_ experience

### DISCUSSION THREAD

**Discussion by losnappas** [jj-vcs/jj#4744] [Ideas]
I am constantly doing `jj st` and `jj log`, then getting into `less`, having to `q` out and losing context, forgetting everything immediately, then having to `jj st/log/w.e` _again_ and looking at the hashes again etc. etc.

I wish the pager picker was a bit smarter ootb. When I do `jj st`, I don't need a full-blown `less` for the 5 lines of text, what I want is `less -X` or `more` or something. Maybe this is PEBCAK situation, idk, I just had to mention it. Possibly a dupe too, I felt super lazy, sorry.

**Comment by ocharles** [comment]
Try adding this to your config:

```
      [ui]
      pager="less -SFRX"
```

**Reply by losnappas** [reply]
Initial impression: better for sure, thanks. Didn't try with longer logs yet though

**Comment by johanneskastl** [comment]
I would rephrase this a little:

I would like to see the pager being configurable for different commands. Currently it is either `auto` or `never` or a user-defined command.

I would like to have `jj diff` user a pager, but not `jj st` or `jj log`. AFAIK this is not yet configurable. Is this worth a feature request?

Kind Regards,
Johannes

**Reply by PhilipMetzger** [reply]
yes, file a FR for it

**Reply by johanneskastl** [reply]
Done, https://github.com/jj-vcs/jj/issues/5217
